### PR TITLE
isLoggedIn

### PR DESCRIPTION
**Pull request by jwolfcrafter**
Changes in app.js to set up a function that will check every route and redirect you to home if you aren't logged in
Changes in users.js to:
-not make the password, even hashed, to ever appear in the stored cookie
-destroy the cookie on user deletion

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -41,6 +42,35 @@ app.use(rewriteUnsupportedBrowserMethods);
...
+  if(req.session.user) loggedin = true;
+  //Logs some useful stuff to the console, can be commented out
+  console.log(`[${date}]: ${reqmethod} ${reqroute} | Authorized: ${loggedin}`);
+  //if you're logged in, proceed to whatever page you were trying to access
+  if(loggedIn) {
+    next();
+  }
+  //if you aren't logged in, you're redirected to the home page, never even hitting the page you were trying to hit
+  else {
+    res.redirect('/');
```
Comment (mpolimen) [inline src/app.js:69]: I say we do 403 forbidden page instead

Comment (jwolfcrafter) [reply src/app.js]: Sounds good, wasn't sure how we wanted to do that

File: src/routes/users.js [hunk 2]
```diff
@@ -81,6 +84,7 @@ router.post('/login', async (req, res) => {
 router.post('/delete', async (req, res) => {
```
[the remark this answers is not in the document]
Comment (jwolfcrafter) [reply src/routes/users.js:96]: Done

**Comment by mpolimen** [comment]
Sorry I didn't think of this before, but if we are posting to /user/login and /user/signup with credentials to set the cookie. The middleware will redirect to the landing page whenever someone tries to log in or sign up. Can you possible make an exception for these two routes within the middleware?

**Comment by mpolimen** [comment]
Looks fancy, I'll drop a "reviewed once" on trello

**Comment by roccopolimen** [comment]
LGTM, merging

**Pull request merged**